File: content/examples/other/leetcode/problems/two-sum/example-2/Solution.java
import java.util.*;

public class Solution {
  public static int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> seen = new HashMap<>();
    for (int i = 0; i < nums.length; i++) {
      int need = target - nums[i];
      if (seen.containsKey(need)) return new int[]{seen.get(need), i};
      seen.put(nums[i], i);
    }
    return null;
  }

  public static void main(String[] args) {
    System.out.println(Arrays.toString(twoSum(new int[]{2,7,11,15}, 9)));
  }
}

File: content/examples/other/leetcode/problems/two-sum/example-2/solution.js
function twoSum(nums, target) {
  const seen = new Map();
  for (let i = 0; i < nums.length; i += 1) {
    const need = target - nums[i];
    if (seen.has(need)) return [seen.get(need), i];
    seen.set(nums[i], i);
  }
  return null;
}

if (require.main === module) {
  console.log(twoSum([2, 7, 11, 15], 9));
}

module.exports = { twoSum };
